README.md

```diff
@@ -8,9 +8,113 @@ This library works with both plain JS as well as AngularJS applications.
 
-##  Usage
 
-You can learn in detail about ADAL JS installation and usage documented in the [ADAL Basics Wiki](https://github.com/AzureAD/azure-activedirectory-library-for-js/wiki/ADAL-basics).
+## Installation
+
+You have multiple ways of getting ADAL JS:
+
+Via NPM (https://www.npmjs.com/package/adal-angular):
+
+    npm install adal-angular
+
+*Note:* Currently there is one NPM package providing both the plain JS library (adal.js) and the AngularJS wrapper (adal-angular.js).
+
+Via CDN:
+
+    <!-- Latest compiled and minified JavaScript -->
+    <script src="https://secure.aadcdn.microsoftonline-p.com/lib/1.0.17/js/adal.min.js"></script>
+    <script src="https://secure.aadcdn.microsoftonline-p.com/lib/1.0.17/js/adal-angular.min.js"></script>
+
+Via Bower:
+
+    $ bower install adal-angular
+
+## Usage
+
+#### Prerequisite
+
+Before using ADAL JS, follow the instructions to [register your application](https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-integrating-applications) on the Azure portal. Also, make sure to enable the [OAuth 2.0 implicit flow](https://docs.microsoft.com/en-us/azure/active-directory/develop/v1-oauth2-implicit-grant-flow) by setting the property `oauth2AllowImplicitFlow` to true by editing your application manifest on the portal. Implicit flow is used by ADAL JS to get tokens.
+
+#### 1. Instantiate the AuthenticationContext
+
+Instantiate the global variable AuthenticationContext with a minimal required config of clientID. You can read about other configurable options [here](https://github.com/AzureAD/azure-activedirectory-library-for-js/wiki/Config-authentication-context#configurable-options).
+
+```JavaScript
+window.config = {
+    clientId: '[Enter your client_id here, e.g. g075edef-0efa-453b-997b-de1337c29185]',
+    popUp: true
+};
+
+var authContext = new AuthenticationContext(config);
+```
+
+#### 2. Login the user
+
+Your app must login the user to establish user context. The login operates in popup mode if you set the option `popUp: true` as shown in the config above.
+
+```JavaScript
+var user = authenticationContext.getCachedUser();
+if (user) {
+    // Use the logged in user information
+    onLogin(null, user);
+}
+else {
+    // Initiate login
+    authenticationContext.login();
+}
+```
+
+#### 3. Get an access token
+
+Next, you can get access tokens for the APIs your app needs to call using the acquireToken method which attempts to acquire token silently. The acquireToken method takes a callback function as shown below.
+
+If the silent token acquisition fails for some reasons such as expiration, you will need to invoke one of the interactive methods to acquire tokens.
```

Comment:
can you pls clarify something like,
"such as expired session" instead of "such as expiration"

Reply:
Updated